addOns/ascanrules/src/main/java/org/zaproxy/zap/extension/ascanrules/PathTraversalScanRule.java

```diff
@@ -674,15 +689,29 @@ private static class DirNamesContentsMatcher implements ContentsMatcher {
 
         @Override
         public String match(String contents) {
-            if (contents.contains("etc") && contents.contains("bin") && contents.contains("boot")) {
-                Pattern nixDoubleCheckPattern = Pattern.compile("\\betc\\b");
-                Matcher nixDoubleCheckMatcher = nixDoubleCheckPattern.matcher(contents);
+            String result = matchNixDirectories(contents);
+            if (result != null) {
+                return result;
+            }
+            return matchWinDirectories(contents);
+        }
 
-                if (nixDoubleCheckMatcher.find()) {
-                    return "etc";
-                }
+        private String matchNixDirectories(String contents) {
+            Pattern binPattern = Pattern.compile("<a href=\".*?/?bin/?\">bin/?</a>");
+            Pattern etcPattern = Pattern.compile("<a href=\".*?/?etc/?\">etc/?</a>");
+            Pattern bootPattern = Pattern.compile("<a href=\".*?/?boot/?\">boot/?</a>");
```

Comment:
Are they always linked like this?

The pattern seems strange having both * and ? side by side. They should probably also be case insensitive.

Reply:
In those cases that I managed to find - yes. But doubts are present, it is not often possible to meet such a situation. I will try to offer an alternative solution.

Reply:
I wasn't saying it's wrong (that we're assuming they're linked like that), I just haven't done any research to ensure it

Reply:
I saw two cases:
1) "<td><a href=\"/bin/\">bin</a></td>" ...
2) "<li><a href="bin/">bin/</a></li>" ...
In both cases, there was a structure with href. And in different cases, slashes are different. When I drew up ideas for regex, I was guided by this logic